src/test/scala/eventstore/j/EsConnectionITest.scala

```diff
@@ -165,7 +165,7 @@ class EsConnectionITest extends eventstore.util.ActorSpec {
 
     "publish stream events" in new TestScope {
       val streamId = s"java-publish-$randomUuid"
-      val publisher = connection.streamPublisher(streamId, null, false, null, false)
+      def publisher = connection.streamPublisher(streamId, null, false, null, false)
```

Comment:
This is because `publisher` immediately starts talking with ES and it is started as finite, hence it completes before we have written events to ES.